src/Tribe/Main.php

```diff
@@ -2272,6 +2272,157 @@ public function eventQueryVars( $qvars ) {
+		 *
+		 * @param  string $method Use "regex" to return a Regular Expression with the possible Base Slugs using l10n
+		 * @return object         Return Base Slugs with l10n variations
+		 */
+		public function get_rewrite_base_slugs( $method = 'regex' ) {
+			$bases = apply_filters( 'tribe_events_rewrite_base_slugs', array(
+				'month' => (array) $this->monthSlug,
+				'list' => (array) $this->listSlug,
+				'today' => (array) $this->todaySlug,
+				'day' => (array) $this->daySlug,
+				'tag' => (array) 'tag',
+				'tax' => (array) 'category',
+				'page' => (array) 'page',
+				'all' => (array) 'all',
+				'singular' => (array) $this->getOption( 'singleEventSlug', 'event' ),
+				'plural' => (array) $this->getOption( 'eventsSlug', 'events' ),
+			) );
+
+			$languages = array( get_locale() );
+
+			$domains = array(
+				'default' => true, // Default doesn't need file path
+				'tribe-events-calendar' => $this->pluginDir . 'lang/',
+			);
+
+			// If WPML exists we treat the multiple languages
+			if ( function_exists( 'icl_get_languages' ) ) {
+				global $sitepress;
+
+				// Grab all languages
+				$_languages = $sitepress->get_active_languages();
+
+				// Query the Current Language
+				$clang = $sitepress->get_current_language();
+				$clocale = $sitepress->get_locale( $clang );
+
+				foreach ( $_languages as $lang ) {
+					$languages[] = $sitepress->get_locale( $lang['code'] );
+				}
+
+				// Prevent Duplicates and Empty langs
+				$languages = array_filter( array_unique( $languages ) );
+
+				// Get the strings on multiple Domains and Languages
+				$bases = $this->get_i18n_strings( $bases, $languages, $domains );
+			}
+
+			if ( 'regex' === $method ){
+				foreach ( $bases as $type => $base ) {
+					$base = array_map( 'sanitize_title_with_dashes', $base );
+					$bases[ $type ] = '(' . implode( '|', $base ) . ')';
+				}
+			}
+
+			return (object) $bases;
+		}
+
+		/**
+		 * Get all possible translations for a String based on the given Languages and Domains
+		 *
+		 * WARNING: This function is slow because it deals with files, so don't overuse it!
+		 *
+		 * @todo Include support for the `load_theme_textdomain` + `load_muplugin_textdomain`
+		 *
+		 * @param  array  $strings          An array of strings (required)
+		 * @param  array  $languages        Which l10n to fetch the string (required)
+		 * @param  array  $domains          Possible Domains to re-load
+		 * @param  string $default_language The default language to avoid re-doing that
+		 *
+		 * @return array                    A multi level array with the possible translations for the given strings
+		 */
+		public function get_i18n_strings( $strings, $languages, $domains = array( 'tribe-events-calendar' ), $default_language = 'en_US' ) {
+			foreach ( $languages as $language ) {
+				// Prevent re-loading the default language
+				if ( $language === $default_language ){
+					continue;
+				}
+
+				foreach ( (array) $domains as $domain => $file ) {
```

Comment:
I'm curious about `$domain => $file` and how it relates to the default `$domains` variable in the method declaration. The default declaration just has a value and is lacking an index of note that would relate to `$domain`.